test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py

```diff
@@ -2139,46 +2074,14 @@ def validate(self):
 
             # See if current version => deprecated.removed_in, ie, should be docs only
-            if isinstance(doc_info['ANSIBLE_METADATA']['value'], ast.Dict) and 'removed' in ast.literal_eval(doc_info['ANSIBLE_METADATA']['value'])['status']:
-                end_of_deprecation_should_be_removed_only = True
-            elif docs and 'deprecated' in docs and docs['deprecated'] is not None:
-                end_of_deprecation_should_be_removed_only = False
-                if 'removed_at_date' in docs['deprecated']:
-                    try:
-                        removed_at_date = docs['deprecated']['removed_at_date']
-                        if parse_isodate(removed_at_date) < datetime.date.today():
-                            msg = "Module's deprecated.removed_at_date date '%s' is before today" % removed_at_date
-                            self.reporter.error(
-                                path=self.object_path,
-                                code='deprecated-date',
-                                msg=msg,
-                            )
-                    except ValueError:
-                        # Already checked during schema validation
-                        pass
-                if 'removed_in' in docs['deprecated']:
-                    try:
-                        collection_name, version = self._split_tagged_version(docs['deprecated']['removed_in'])
-                        if collection_name != self.collection_name:
-                            self.reporter.error(
-                                path=self.object_path,
-                                code='invalid-module-deprecation-source',
-                                msg=('The deprecation version for a module must be added in this collection')
-                            )
-                            # Treat the module as not to be removed:
-                            raise ValueError('')
-                        removed_in = self.StrictVersion(str(version))
-                    except ValueError:
-                        end_of_deprecation_should_be_removed_only = False
-                    else:
-                        if not self.collection:
-                            strict_ansible_version = self.StrictVersion('.'.join(ansible_version.split('.')[:2]))
-                            end_of_deprecation_should_be_removed_only = strict_ansible_version >= removed_in
-                        elif self.collection_version:
-                            strict_ansible_version = self.collection_version
-                            end_of_deprecation_should_be_removed_only = strict_ansible_version >= removed_in
-                        else:
-                            end_of_deprecation_should_be_removed_only = False
+            if docs and 'deprecated' in docs and docs['deprecated'] is not None:
+                try:
+                    removed_in = StrictVersion(str(docs.get('deprecated')['removed_in']))
```

Comment:
The old code checked both `removed_in` and `removed_at_date`. Also it did handle tagged versions, and understood collection versions vs. Ansible versions.

Reply:
I think this should simply be
```.py
            if docs and 'deprecated' in docs and docs['deprecated'] is not None:
                end_of_deprecation_should_be_removed_only = False
                if 'removed_at_date' in docs['deprecated']:
                    try:
                        removed_at_date = docs['deprecated']['removed_at_date']
                        if parse_isodate(removed_at_date) < datetime.date.today():
                            msg = "Module's deprecated.removed_at_date date '%s' is before today" % removed_at_date
                            self.reporter.error(
                                path=self.object_path,
                                code='deprecated-date',
                                msg=msg,
                            )
                    except ValueError:
                        # Already checked during schema validation
                        pass
                if 'removed_in' in docs['deprecated']:
                    try:
                        collection_name, version = self._split_tagged_version(docs['deprecated']['removed_in'])
                        if collection_name != self.collection_name:
                            self.reporter.error(
                                path=self.object_path,
                                code='invalid-module-deprecation-source',
                                msg=('The deprecation version for a module must be added in this collection')
                            )
                            # Treat the module as not to be removed:
                            raise ValueError('')
                        removed_in = self.StrictVersion(str(version))
                    except ValueError:
                        end_of_deprecation_should_be_removed_only = False
                    else:
                        if not self.collection:
                            strict_ansible_version = self.StrictVersion('.'.join(ansible_version.split('.')[:2]))
                            end_of_deprecation_should_be_removed_only = strict_ansible_version >= removed_in
                        elif self.collection_version:
                            strict_ansible_version = self.collection_version
                            end_of_deprecation_should_be_removed_only = strict_ansible_version >= removed_in
                        else:
                            end_of_deprecation_should_be_removed_only = False
```
i.e. everything that was removed except the first `if` branch on ANSIBLE_METADATA.